### PR TITLE
Allow init without hacker address

### PR DESCRIPTION
**Pull request by ben-kaufman**

### CODE REVIEW

File: contracts/EthicalReturn.sol [hunk 1]
```diff
@@ -39,8 +43,25 @@ contract EthicalReturn is ReentrancyGuard {
...
+            revert MustHaveHackerBeforeDeposit();
+        } 
+    }
 
-    receive() external payable {}
+    function deposit(address _hacker) external payable {
+        if (_hacker == address(0)) {
+            revert InvalidHacker();
+        }
+        if (hacker != address(0)) {
```
Comment (jellegerbrandy) [inline contracts/EthicalReturn.sol:57]:
```suggestion
        if (hacker != address(0) && _hacker != hacker) 
```

Comment (jellegerbrandy) [reply contracts/EthicalReturn.sol:57]: In this way the deposit function can be used twice

**Review by jellegerbrandy — CHANGES_REQUESTED** [review]
see the comment

File: contracts/EthicalReturn.sol (hunk 1 above)
Comment (ben-kaufman) [reply contracts/EthicalReturn.sol:57]: Implemented a bit differently so the minimum amount isn't checked on extra deposits and the hacker isn't unnecessarily set.

Comment (jellegerbrandy) [reply contracts/EthicalReturn.sol:57]: nice

**Review by jellegerbrandy — APPROVED** [review]

**Pull request merged**